File: fixtures/scale/javascript/large/src/checks/checkStaticpad.js
import staticpad from '../ciphers/staticpad.js';
import staticpadSpecMod from '../specs/staticpadSpec.js';
import corpusMaskMod from '../corpus/corpusMask.js';

function checkStaticpad() {
  const spec = staticpadSpecMod.staticpadSpec();
  for (const text of corpusMaskMod.corpusMask()) {
    const sealed = staticpad.staticpadEncrypt(text, spec.key);
    const opened = staticpad.staticpadDecrypt(sealed, spec.key);
    if (opened !== text) return false;
  }
  return spec.category === 'mask';
}

export default { checkStaticpad };

File: fixtures/scale/lua/large/checks/check_staticpad.lua
local staticpad = require("ciphers.staticpad")
local staticpad_spec = require("specs.staticpad_spec")
local corpus_mask = require("corpus.corpus_mask")

local function check_staticpad()
  local spec = staticpad_spec.get()
  for _, text in ipairs(corpus_mask.corpus_mask()) do
    local sealed = staticpad.staticpad_encrypt(text, spec.key)
    local opened = staticpad.staticpad_decrypt(sealed, spec.key)
    if opened ~= text then return false end
  end
  return spec.category == "mask"
end

return { check_staticpad = check_staticpad }
